tests/test_custom_attributes.py

```diff
@@ -0,0 +1,162 @@
+GET_RETURN_VALUES = {
+    'no_cas': {},
+    'ca_exist': {
+        "custom_attributes": [{
+            "href": "{miq_hostname}/api/providers/{provider_id}/custom_attributes/{ca_id}".format(miq_hostname=MANAGEIQ_HOSTNAME, provider_id=PROVIDER_ID, ca_id=EXISTING_CA['id']),
+            "id": EXISTING_CA['id'],
+            "section": "metadata",
+            "name": EXISTING_CA['name'],
+            "value": EXISTING_CA['value'],
+            "resource_type": "ExtManagementSystem",
+            "resource_id": PROVIDER_ID,
+            "source": "EVM",
+            "serialized_value": EXISTING_CA['value']
+        }]
+    }
+}
+POST_RETURN_VALUES = {
+    'added_ca': {
+        'results': [{
+            "resource_id": PROVIDER_ID,
+            "section": "metadata",
+            "name": NEW_CA['name'],
+            "id": NEW_CA['id'],
+            "serialized_value": NEW_CA['value'],
+            "value": NEW_CA['value'],
+            "resource_type": "ExtManagementSystem",
+            "source": "EVM"
+        }]
+    },
+    'updated_ca': {
+        'results': [{
+            "resource_id": PROVIDER_ID,
+            "section": "metadata",
+            "name": EXISTING_CA['name'],
+            "id": EXISTING_CA['id'],
+            "serialized_value": UPDATED_CA_VALUE,
+            "value": UPDATED_CA_VALUE,
+            "resource_type": "ExtManagementSystem",
+            "source": "EVM"
+        }]
+    }
+}
+
+
+@pytest.fixture(autouse=True)
+def miq_api_class(monkeypatch):
+    miq_api_class = Mock(spec=API)
+    monkeypatch.setattr("manageiq_custom_attributes.MiqApi", miq_api_class)
+    yield miq_api_class
+
+
+@pytest.fixture
+def the_provider():
+    the_provider = Mock()
+    the_provider.name = PROVIDER_NAME
+    yield the_provider
+
+
+@pytest.fixture
+def miq_ansible_module():
+    miq_ansible_module = Mock(spec=AnsibleModule)
+    yield miq_ansible_module
+
+
+class AnsibleModuleFailed(Exception):
+    pass
+
+
+@pytest.fixture()
+def miq(miq_api_class, miq_ansible_module, the_provider):
+    # miq_ansible_module.params = {'metrics': True}
+
+    def fail(msg):
+        raise AnsibleModuleFailed(msg)
+
+    miq_ansible_module.fail_json = fail
+    miq = manageiq_custom_attributes.ManageIQ(miq_ansible_module, MANAGEIQ_HOSTNAME,
+                                              "The username", "The password")
+
+    miq_api_class.return_value.collections.providers = [the_provider]
+
+    yield miq
+
+
+def test_get_entity_cas(miq, miq_api_class):
+    miq_api_class.return_value.get.return_value = GET_RETURN_VALUES['ca_exist']
+    provider_cas = miq.get_entity_cas('provider', PROVIDER_ID)
+    assert provider_cas == [{
+      "href": "http://miq.example.com/api/providers/266/custom_attributes/6226",
+      "id": 6226,
+      "section": "metadata",
+      "name": "ca1",
+      "value": "value 1",
+      "resource_type": "ExtManagementSystem",
+      "resource_id": 266,
+      "source": "EVM",
+      "serialized_value": "value 1"
+    }]
```

Comment:
I might be wrong because I am not used to pytest but won't it be easier and more clear to compare to `GET_RETURN_VALUES['ca_exist']`?